src/metabase-report-notifier/src/main/java/com/champtitles/metabasereportexecutor/notifier/App.java

```diff
@@ -2,15 +2,114 @@
 
 import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.RequestHandler;
-import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.lambda.runtime.events.SNSEvent;
+import com.champtitles.metabasereportexecutor.executor.KmsDecrypt;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class App implements RequestHandler<S3Event, Void> {
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.NoSuchElementException;
+
+public class App implements RequestHandler<SNSEvent, Void> {
+
+    private static final Logger logger = LoggerFactory.getLogger(App.class.getName());
+    private static final String awsRegion = System.getenv("AWS_REGION");
+    private static final String bucket = System.getenv("BUCKET");
+    private static final String smtpHost = System.getenv("SMTP_HOST");
+    private static final String smtpPort = System.getenv("SMTP_PORT");
+    private static final String smtpUser = System.getenv("SMTP_USER");
+    private static final String smtpPasswordKms = System.getenv("SMTP_PASSWORD_KMS");
+    private static final String fromAddress = System.getenv("FROM_ADDRESS");
+    private static final String recipients = System.getenv().getOrDefault("RECIPIENTS", "test@example.com");
+    private static final String metabaseCardId = System.getenv().getOrDefault("METABASE_CARD_ID", "1");
```

Comment:
Is this a valid default?

Reply:
@dcrough-ownum not valid at all actually. It was the shortest path to set the variable for unit testing. I can certainly refactor it to set the values differently but I was still looking for patterns for best practice.

Reply:
There may be a better pattern overall, where all of these are passed in via a constructor. Let something else resolve the env vars and when this is constructed pass them in (preferably in an object), then you can construct this parameter object as needed in unit tests.

Reply:
@dcrough-ownum I agree, a "config loader" type of class coupled with a model of config values and that gets passed to the constructor. That would be much better than a dozen constants at the top of the class for sure. I created an issue to track the refactoring: #28